public_html/wp-content/plugins/pattern-directory/includes/pattern-validation.php

```diff
@@ -200,5 +204,141 @@ function validate_status( $prepared_post, $request ) {
+	if ( 'publish' !== $target_status ) {
+		return $prepared_post;
+	}
+
+	// Extract strings and URLs, run against spam checks.
+	$post = get_post( $prepared_post->ID );
+
+	$title       = $prepared_post->post_title ?? $post->post_title;
+	$content     = $prepared_post->post_content ?? $post->post_content;
+	$description = $request['meta']['wpop_description'] ?? ( $post->wpop_description ?: '' );
+	$keywords    = $request['meta']['wpop_keywords'] ?? ( $post->wpop_keywords ?: '' );
+
+	// Extract URLs.
+	$links = array();
+	if ( preg_match_all( '![\b\W"\'](?P<link>((http|https|ftp|mailto):)?//[/.\w]+)[\b\W"\']!', $content, $m ) ) {
+		$links = array_unique( $m['link'] );
+	}
+
+	// Stringify.
+	if ( ! class_exists( '\WordPressdotorg\Pattern_Translations\Pattern' ) ) {
+		// This is just a fall-back for local environments where the Translator isn't active.
+		// not designed to be used in production.
+		$strings = array(
+			$title,
+			$description,
+			wp_strip_all_tags( $content ),
+			$keywords,
+		);
+	} else {
+		$pattern              = new Translations_Pattern();
+		$pattern->ID          = $post->ID;
+		$pattern->title       = $title;
+		$pattern->name        = $post->post_name;
+		$pattern->description = $description;
+		$pattern->keywords    = $keywords;
+		$pattern->html        = $content;
+		$pattern->locale      = get_locale();
+
+		$parser  = new Translations_PatternParser( $pattern );
+		$strings = $parser->to_strings();
+	}
+
+	// Combine strings for ease of use.
+	$combined_strings = implode( "\n", $strings );
+	$combined_links   = implode( "\n", $links );
+	$combined         = $combined_strings . "\n" . $combined_links;
+
+	// Not yet detected as spam.
+	$is_spam = false;
+
+	// Treat Paragraph-only submissions as likely spam.
+	if ( ! $is_spam ) {
+		// Only fetches the top-level of blocks, we're only
+		$block_names_in_use = array_filter(
+			array_unique(
+				wp_list_pluck(
+					parse_blocks( $content ),
+					'blockName'
+				)
+			)
+		);
+
+		if ( array( 'core/paragraph' ) === $block_names_in_use ) {
+			$is_spam = true;
+		}
+	}
+
+	// Run it past Akismet.
+	if ( ! $is_spam && is_callable( array( 'Akismet', 'rest_auto_check_comment' ) ) ) {
+		$current_user = wp_get_current_user();
+
+		$akismet_payload = array(
+			'comment_post_ID'      => 0,
+			'comment_type'         => 'pattern_submission',
+			// Disabled as logged in users get bonus points I think, which we don't want.
+			// 'user_ID'           => get_current_user_id(),
+			'comment_author'       => $current_user->display_name ?: $current_user->user_login,
+			'comment_author_email' => $current_user->user_email,
+			'comment_author_url'   => '',
+			'comment_content'      => $combined_strings,
+			'comment_content_raw'  => $content,
+			'permalink'            => get_permalink( $post ),
+		);
+
+		$akismet = \Akismet::rest_auto_check_comment( $akismet_payload );
+		if ( is_wp_error( $akismet ) ) {
+			$akismet = array( 'akismet_result' => 'discard' );
+		}
```

Comment:
No.

This is some uglyness from the Akismet plugin API not being designed to be used outside of comments.

If the "Discard the worst of the worst spam" is enabled, Akismet will return a WP_Error instead of changing the status to spam when it thinks it's being called from the REST API (that's why `rest_auto_check_comment` is called). If it thinks it's being called outside of the rest-api then it redirects and/or die's.